sumpy/expansion/__init__.py

```diff
@@ -442,6 +441,43 @@ def stored_identifiers(self):
         stored_identifiers, _ = self.get_stored_ids_and_unscaled_projection_matrix()
         return stored_identifiers
 
+    @memoize_method
+    def _get_mi_ordering_key(self):
+        """Calculate the multi-index that appears last in the PDE given the pde_dict
+        A degree lexicographic order with the slowest varying index depending on
+        the PDE is used. For two dimensions, this is either deglex or degrevlex.
+        """
+        dim = self.dim
+        pde_dict, = self.knl.get_pde_as_diff_op().eqs
+        slowest_varying_index = 0
+        for ident in pde_dict.keys():
+            if ident.mi.count(0) == dim - 1:
+                non_zero_index = next(i for i in range(self.dim) if ident.mi[i] != 0)
+                slowest_varying_index = max(slowest_varying_index, non_zero_index)
+
+        mi_compare_axis = list(range(self.dim))
+        mi_compare_axis[0], mi_compare_axis[slowest_varying_index] = \
+                slowest_varying_index, 0
```

Comment:
It looks like you're computing an axis permutation here. Maybe name it as such?

Reply:
Done in 90e77d0